vendor/bat-native-ads/src/bat/ads/internal/ml/pipeline/pipeline_util.cc

```diff
@@ -183,9 +188,11 @@ absl::optional<model::Linear> ParsePipelineClassifier(
 
 }  // namespace
 
-absl::optional<PipelineInfo> ParsePipelineJSON(const std::string& json) {
+absl::optional<PipelineInfo> ParsePipelineJSON(std::string json) {
   absl::optional<base::Value> root = base::JSONReader::Read(json);
 
+  json = std::string();  // Free memory, because |json| can be ~10Mb.
```

Comment:
Do we really need this? As memory will be freed on function exit

Reply:
Yep, we need this. In the middle of function we allocate a lot of an extra memory. If we don't drop `json` here the peak size will be bigger.
But this doesn't change the final size.

Reply:
I've improved the comment to clarify things.